repoupdater.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import glob
+import os
+import re
+import sys
+import subprocess
+
+from github import Github
+from github.GithubException import GithubException
+import yaml
+
+import client
+
+ORG_NAME = "OpenSAFELY"
+REPO_NAME_PATTERN = "^.*-research$"
+BASE_PATH = os.path.abspath("research")
+
+
+def main():
+    parser = argparse.ArgumentParser(prog="repoupdater")
+    subparsers = parser.add_subparsers(help="sub-command help", dest="subcommand")
+    list_parser = subparsers.add_parser("list", help="list all repos")
+    update_parser = subparsers.add_parser(
+        "update", help="update all repos via pull or clone"
+    )
+    exec_parser = subparsers.add_parser(
+        "exec", help="execute command against all repos"
+    )
+    exec_parser.add_argument("command")
+    exec_parser.add_argument("args", nargs="*")
+    pull_request_parser = subparsers.add_parser(
+        "pull-request", help="submit pull request against each repo"
+    )
+    pull_request_parser.add_argument("branch")
+    pull_request_parser.add_argument("title")
+
+    args = parser.parse_args()
+
+    if args.subcommand == "list":
+        list_()
+    elif args.subcommand == "update":
+        update()
+    elif args.subcommand == "exec":
+        exec_([args.command] + args.args)
+    elif args.subcommand == "pull-request":
+        pull_request(args.branch, args.title)
+    else:
+        assert False, args.subcommand
+
+
+def list_():
+    client = get_client()
+    for repo in get_repos(client):
+        print(repo.html_url)
+
+
+def update():
+    client = get_client()
+    repos = get_repos(client)
+    if check_for_uncommitted_changes(repos):
+        sys.exit(1)
+
+    for repo in repos:
+        path = os.path.join(BASE_PATH, repo.name)
+        print("-" * 80)
+        print(repo.name)
+
+        if os.path.exists(path):
+            os.chdir(path)
+            subprocess.run(["git", "checkout", "master"], check=True)
+            subprocess.run(["git", "pull"], check=True)
+        else:
+            subprocess.run(["git", "clone", repo.ssh_url, path], check=True)
+
+
+def exec_(argv):
+    for path in sorted(glob.glob(os.path.join(BASE_PATH, "*"))):
```

Comment:
ah, ok, it's not all repos, it's all subdirectories, which is a bit counterintuitive to me. Would it be better to do the equivalent of https://github.com/opensafely/sysadmin/pull/7/files#diff-fdb7becf8fb97e0422febce02a3042d9a1d3db6539ed13d30f53f971c9c65989R63-R64 here?